### PR TITLE
feat(pagination): upgrade to v3

### PR DESCRIPTION
**Pull request by thisisandy**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/vc-pagination/Pager.jsx [hunk 1]
```diff
@@ -26,18 +27,24 @@ export default {
...
+    );
 
     return (
       <li
-        class={cls}
         onClick={this.handleClick}
         onKeypress={this.handleKeyPress}
         title={this.showTitle ? this.page : null}
         tabIndex="0"
+        {...this.$attrs}
```
Comment (tangjinzhou) [inline components/vc-pagination/Pager.jsx:46]: 这种写法，onClick onKeypress 会触发两次，jsx内部的 mergeProps 会合并事件

Comment (tangjinzhou) [reply components/vc-pagination/Pager.jsx:46]: @Amour1688 也看看   jsx是否应该是替换  而不是 合并

File: components/vc-pagination/Pagination.jsx [hunk 2]
```diff
@@ -359,8 +361,12 @@ export default {
       }
       const hasPrev = this.hasPrev();
       const hasNext = this.hasNext();
+
       return (
-        <ul class={`${prefixCls} ${prefixCls}-simple`}>
+        <ul
+          {...this.$attrs}
```
Comment (tangjinzhou) [inline components/vc-pagination/Pagination.jsx:367]: 这么写会把所有的事件也绑定到 ul 上

File: components/vc-pagination/Pagination.jsx [hunk 3]
```diff
@@ -580,12 +582,16 @@ export default {
...
     const prevDisabled = !this.hasPrev() || !allPages;
     const nextDisabled = !this.hasNext() || !allPages;
-    const buildOptionText = this.buildOptionText || this.$scopedSlots.buildOptionText;
+    const buildOptionText = this.buildOptionText || this.$slots.buildOptionText?.();
     return (
       <ul
-        class={{ [`${prefixCls}`]: true, [`${prefixCls}-disabled`]: disabled }}
         unselectable="unselectable"
         ref="paginationNode"
+        {...this.$attrs}
```
Comment (tangjinzhou) [inline components/vc-pagination/Pagination.jsx:590]: 同样  而且 这里你的 class 会出现重复的挂载吧

File: components/vc-pagination/Pager.jsx (hunk 1 above)
Comment (Amour1688) [reply components/vc-pagination/Pager.jsx:46]: 这种写法，在运行时 mergeProps 会把事件给合并

**Comment by tangjinzhou** [comment]
https://github.com/vueComponent/ant-design-vue/commit/2ff8ba05a4264283c995c1277e9f9d4e80116e9b 
最基本的事例要测试的呀  😓

**Pull request merged**